### PR TITLE
[14.0][MIG] hr_timesheet_sheet_period

### PR DESCRIPTION
**Pull request by JuanyDForgeflow**
This module allows to create timesheets with start and end dates matching with the
payroll period of the schedule pay that is specified in the contract of the
employee or if not then will match the nearest payroll period.

### CODE REVIEW

**Comment by alexey-pelykh** [comment]
@JuanyDForgeflow please rebase it

**Comment by JuanyDForgeflow** [comment]
@alexey-pelykh Test fail after removing travis, I added this as rebel model in .copier-answers.yml but it still doesn't work, any suggestions?

**Comment by alexey-pelykh** [comment]
I guess there's really something wrong with the PR itself and some tests need to be adapted to changes of 14

**Comment by alexey-pelykh** [comment]
I guess there's really something wrong with the PR itself and some tests need to be adapted to changes of 14

**Comment by JuanyDForgeflow** [comment]
@alexey-pelykh but the test are failing for hr_utilization_report module not this one, should I change that test here?

**Comment by alexey-pelykh** [comment]
yet the tests don't fail in the 14 branch, so maybe this PR somehow affects the thing. Will look into that as well

**Comment by alexey-pelykh** [comment]
@JuanyDForgeflow could you do the honors of rebasing please :)

**Comment by JuanyDForgeflow** [comment]
@alexey-pelykh I tried rebasing but it's up to date, I rebased yesterday

**Comment by alexey-pelykh** [comment]
@JuanyDForgeflow the 14 branch changed

**Comment by alexey-pelykh** [comment]
Because of #531

**Comment by JuanyDForgeflow** [comment]
@alexey-pelykh it's rebased now

**Comment by alexey-pelykh** [comment]
@JuanyDForgeflow now it's your turn, the error is different

File: .travis.yml [hunk 1]
```diff
@@ -1,50 +0,0 @@
-language: python
```
Comment (alexey-pelykh) [inline .travis.yml]: I don't think it needs to be removed since travis is used in forks?

File: hr_timesheet_sheet_period/models/res_company.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+SHEET_RANGE_PAYROLL_PERIOD = 100
+
+
+class ResCompany(models.Model):
+    _inherit = "res.company"
+
+    sheet_range = fields.Selection(
+        selection_add=[
+            (SHEET_RANGE_PAYROLL_PERIOD, "Payroll Period"),
```
Comment (alexey-pelykh) [inline hr_timesheet_sheet_period/models/res_company.py:14]: This causes an issue why builds fail, key needs to be a string https://github.com/OCA/timesheet/blob/14.0/hr_timesheet_sheet/models/res_company.py#L20

File: .copier-answers.yml [hunk 3]
```diff
@@ -8,8 +8,9 @@ include_wkhtmltopdf: false
 odoo_version: 14.0
 org_name: Odoo Community Association (OCA)
 org_slug: OCA
-rebel_module_groups: []
-repo_description: 'TODO: add repo description.'
+rebel_module_groups:
+- hr_timesheet_sheet_period
```
Comment (alexey-pelykh) [inline .copier-answers.yml]: I have no idea why this is needed

File: hr_timesheet_sheet_period/models/res_company.py (hunk 2 above)
Comment (JuanyDForgeflow) [reply hr_timesheet_sheet_period/models/res_company.py:14]: @alexey-pelykh this was it thanks! i missed it completely

**Comment by JuanyDForgeflow** [comment]
@alexey-pelykh it's finally green

File: hr_timesheet_sheet_period/models/res_company.py [hunk 4]
```diff
@@ -3,14 +3,12 @@
 
 from odoo import fields, models
 
-SHEET_RANGE_PAYROLL_PERIOD = 100
```
Comment (alexey-pelykh) [inline hr_timesheet_sheet_period/models/res_company.py]: please keep the constant just make it a string :) otherwise this magic "100" is everywhere

**Review by alexey-pelykh — CHANGES_REQUESTED** [review]

File: hr_timesheet_sheet_period/models/res_company.py (hunk 4 above)
Comment (JuanyDForgeflow) [reply hr_timesheet_sheet_period/models/res_company.py]: @alexey-pelykh done :)

**Review by alexey-pelykh — APPROVED** [review]

**Pull request merged**